it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,312 @@
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.Struct;
+import com.google.common.collect.ImmutableList;
+import java.util.concurrent.ExecutionException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/** Unit tests for {@link DefaultSpannerResourceManager}. */
+@RunWith(JUnit4.class)
+public class DefaultSpannerResourceManagerTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private Spanner spanner;
+
+  @Mock(serializable = true)
+  private ResultSet resultSet;
+
+  @Mock private InstanceAdminClient instanceAdminClient;
+
+  private final String testId = "test-id";
+  private final String projectId = "test-project";
+  private final String region = "us-east1";
+  private DefaultSpannerResourceManager testManager;
+
+  @Captor private ArgumentCaptor<Iterable<Mutation>> writeMutationCaptor;
+  @Captor private ArgumentCaptor<Iterable<String>> statementCaptor;
+  @Captor private ArgumentCaptor<String> instanceIdCaptor;
+  @Captor private ArgumentCaptor<String> databaseIdCaptor;
+
+  @Before
+  public void setUp() {
+    testManager = new DefaultSpannerResourceManager(spanner, testId, projectId, region);
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateInstanceFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getInstanceAdminClient().createInstance(any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
```

Comment:
Make sure to explicitly set up successful mocks, such as for instance creation. You can create a helper method to handle this.

This goes for other tests as well.